packages/wdio-jasmine-framework/src/index.js

```diff
@@ -75,15 +76,23 @@ class JasmineAdapter {
          */
         jasmine.Spec.prototype.addExpectationResult = this.getExpectationResultHandler(jasmine)
 
+        const hookArgsFn = (context) => [{ ...(self.lastTest || {}) }, context]
+
         /**
          * wrap commands with wdio-sync
          */
-        INTERFACES['bdd'].forEach((fnName) => runTestInFiberContext(
-            ['it', 'fit', 'xit'],
-            this.config.beforeHook,
-            this.config.afterHook,
-            fnName
-        ))
+        INTERFACES['bdd'].forEach((fnName) => {
+            const isTest = TEST_INTERFACES.includes(fnName)
+            runTestInFiberContext(
+                isTest,
+                isTest ? this.config.beforeTest : this.config.beforeHook,
+                hookArgsFn,
+                isTest ? this.config.afterTest : this.config.afterHook,
+                hookArgsFn,
```

Comment:
Same args for both hooks? Shouldn't the `afterTest` hook contain the result of the test?

Reply:
It should and it's appended later on, we can't get result at this point.
BeforeTest and AfterTest hooks are executed while test is in progress

Reply:
> It should and it's appended later on

Where does it get appended? Is this a breaking change?

Reply:
It's not a breaking change even though the signature has changed. I'd say it's new feature. 

See 
https://github.com/mgrybyk/webdriverio/blob/ee9a217e3330ddafdf02889a6e0b7530243a28f2/packages/wdio-utils/src/test-framework/testFnWrapper.js

Here is an example for mocha.

Original `it` is replaced by `testFrameworkFnWrapper` that calls not just original `it` but also runs beforeTest before and afterTest after.

We know original `it` result once it is executed in `testFrameworkFnWrapper`, now result can be appended to afterTest args.

Users can use hook as before
```
afterTest(test) {
  if (test.passed) {}
}
```
or in new format (same as afterStep in cucumber works)
```
afterTest(test, context, result) {
  if (result.passed) {}
}
```

Finally we now have same approach to run hooks across all test frameworks and modes (sync, async).